Add the spec v2 draft (#359)
Financial-Times/origami-website: _specification-v2/manifest.md

nav_order: 25
---

# {{page.title}}

`origami.json` is a
<a href="https://www.json.org/"
class="o-typography-link--external">
	<abbr title="JavaScript Object Notation">
		JSON
	</abbr>
</a>
file responsible for describing various aspects of an Origami project.

## Properties

### origami

<table class="o-manifest__table o-table o-table--compact o-table--row-headings o-table--vertical-lines o-table--horizontal-lines" data-o-component="o-table">
	<tr>
		<th scope="row" role="rowheader">Type</th>
		<td><code>String</code></td>
	</tr>
	<tr>
		<th scope="row" role="rowheader">Required</th>
		<td><code>true</code></td>
	</tr>
</table>

The version of Origami to which the project conforms.

**Must** be set to `"2.0"`.

<pre><code class="o-syntax-highlight--json">{
	"origami": "2.0"
}</code></pre>

### type

<table class="o-manifest__table o-table o-table--compact o-table--row-headings o-table--vertical-lines o-table--horizontal-lines" data-o-component="o-table">
	<tr>
		<th scope="row" role="rowheader">Type</th>
		<td><code>String</code> or <code>null</code></td>
	</tr>
	<tr>
		<th scope="row" role="rowheader">Required</th>
		<td><code>true</code></td>
	</tr>
</table>

> Defines the type of Origami project that the manifest belongs to. **Must** be
> set to a string:

Reviewer: we need to mention it an be set to `null`, which is also in the list of possible options
For the code suggestion, I copied the text from the `status` part of this spec which uses "must be one of:"
```suggestion
Defines the type of Origami project that the manifest belongs to. **Must** be one of:
```

Author: oh yes, this was something I wanted to talk about.
I'm wondering if we should say something like:

> **must** be set to a string, .e.g.:

and then remove `null` as an option. `type` then becomes an entry point for extensions. the component spec augments the spec by saying "when the `type` is "component" then all of this applies:" and you opt into this set of rules by setting the type to that

Reviewer: I like that idea a lot.

I checked what sets it to null, and the list is:
- change-api-action -- github action
- origami-apply-labels -- github action
- origami-labels -- github action
- origami-percy -- github action
- origami-project-board-action -- github action
- origami-version -- github action
- polyfill-library-node -- npm libary
- useragent_parser -- npm library

Author: ```suggestion
Defines the type of Origami project that the manifest belongs to. **Must** be set to a string, e.g.:
```

Author: nice work! I've added a description of `"action"` to this list, we can update those to use that. and maybe even we write a spec for our github actions some time:)

- `"component"`: A front-end component that follows [the component specification](/spec/v2/components/)
- `"imageset"`: A set of images that have an alias on the Origami Image Service
- `"service"`: An HTTP service
- `"cli"`:	A command line tool
- `"library"`:	A library that is not a front-end component
- `"website"`: Origami websites that aren't intended to be services
- `"config"`: Projects that are configuration for other projects
- `"example"`: Example and boilerplate projects

> - `"meta"`: Repository-only projects that relate to how Origami works

Author: ```suggestion
- `"action"`: [A Github Action](https://github.com/features/actions)
- `"meta"`: Repository-only projects that relate to how Origami works
```

> - `null`: An Origami project that does not fit any of the named categories

Author: ```suggestion
```
